Bottom: 68027fa27f4e8bb19f5948c96c564e968d131c51
Top:    c269563104dd8f32f395863cc56021a5b5fe4ae3
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-09-08 15:56:25 +0800

Refresh of vhost-speed-up-descriptor-ring

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 955b990..be056fc 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1467,12 +1467,12 @@ long vhost_vring_ioctl(struct vhost_dev *d, int ioctl, void __user *argp)
 		vq->log_addr = a.log_guest_addr;
 		vq->used = (void __user *)(unsigned long)a.used_user_addr;
 
-		int r = get_user_pages_fast((unsigned long)a.desc_user_addr, 1, 1, &vq->page_desc);
+		r = get_user_pages_fast((unsigned long)a.desc_user_addr,
+					1, 1, &vq->page_desc);
 		if (r < 0)
 			return -EFAULT;
 		BUG_ON(r != 1);
-		vq->desc_uaddr = (unsigned long)kmap_atomic(vq->page_desc) +
-			         (unsigned long)vq->desc & (PAGE_SIZE - 1);
+		vq->desc_vaddr = vmap(&vq->page_desc, 1, VM_MAP, PAGE_KERNEL);
 
 		break;
 	case VHOST_SET_VRING_KICK:
diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index f89378d8..7c2d6e6 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -150,7 +150,7 @@ struct vhost_virtqueue {
 #endif
 	u32 busyloop_timeout;
 	struct page *page_desc;
-	void *desc_uaddr;
+	void *desc_vaddr;
 };
 
 struct vhost_msg_node {
